### DISCUSSION TITLE
Getting error when uploading attachments

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#5695] [Help] [answered]
I am getting an validation error when saving an attachment to the row "validationErrors: {image: ["must be of type array"]} 
<img width="833" alt="img-err" src="https://user-images.githubusercontent.com/94033860/166570949-2b6f25db-ff5e-4c08-91ec-11a8f84f0d8d.png">
<img width="776" alt="img-err-1" src="https://user-images.githubusercontent.com/94033860/166570955-0cdbeedc-d9c4-443b-8e5c-9e927fe829fd.png">
"

**Comment by melohagan** [comment]
Hi @newtonlaw95 

This appears to be a bug. In the meantime you can access the attachment via JavaScript instead e.g.

`return $("New Form.Fields.Attachment")`

The handlebars appear to be treating it as a string and not an array. 

Bug raised here: https://github.com/Budibase/budibase/issues/5697

**Reply by melohagan** [reply, marked as answer]
It does work fine however, if you use the table schema and attachment field name.

For example, your table:
![Screenshot 2022-05-04 at 14 30 55](https://user-images.githubusercontent.com/101575380/166691511-54fdfe51-b809-4b42-aa99-f66287d43396.png)

Link your attachment component to the form:
![Screenshot 2022-05-04 at 14 31 25](https://user-images.githubusercontent.com/101575380/166691600-3ec7de87-f68e-477c-ae48-fc90d64a0e2c.png)

Link your form schema to the table that has the attachment column:
![Screenshot 2022-05-04 at 14 32 18](https://user-images.githubusercontent.com/101575380/166691762-e1b63991-b2ad-482c-8d2f-4eff9bdfc15f.png)

Finally just save row for your form:
![Screenshot 2022-05-04 at 14 33 50](https://user-images.githubusercontent.com/101575380/166692069-579373ef-c323-4d1d-9b0e-4ccc1b90177b.png)

**Reply by ghost** [reply]
@melohagan  Thanks for details, tried the same still image is not getting passed in the payload when i do Save action

**Reply by ghost** [reply]
@melohagan This worked by using the button action >> Save row and not mentioning any columns to update / create with handle bars under save row action after choosing the table to save.

**Comment by NicolasFabre81** [comment]
I just had it and the cause was a ' in the picture name !
